CMakeLists.txt.in

```diff
@@ -0,0 +1,215 @@
+cmake_minimum_required(VERSION 3.5)
+project(build-deps NONE)
+
+cmake_host_system_information(
+  RESULT build_concurrency_factor
+  QUERY NUMBER_OF_LOGICAL_CORES)
+set(make_command make -j ${build_concurrency_factor})
+
+include(ExternalProject)
+
+ExternalProject_Add(boost_ext
+  # The 1.67.0 release has a bug in Boost Lockfree around a missing header.
+  URL https://dl.bintray.com/boostorg/release/1.64.0/source/boost_1_64_0.tar.gz
+  URL_MD5 319c6ffbbeccc366f14bb68767a6db79
+  PATCH_COMMAND
+    ./bootstrap.sh
+    --prefix=@SMF_DEPS_INSTALL_DIR@
+    --with-libraries=atomic,chrono,date_time,filesystem,program_options,system,test,thread
+  CONFIGURE_COMMAND ""
+  BUILD_COMMAND ""
+  INSTALL_COMMAND
+    ${CMAKE_COMMAND} -E chdir <SOURCE_DIR>
+    ./b2
+    -j ${build_concurrency_factor}
+    --layout=system
+    --build-dir=<BINARY_DIR>
+    install
+    variant=debug
+    link=shared
+    threading=multi
+    hardcode-dll-paths=true
+    dll-path=@SMF_DEPS_INSTALL_DIR@/lib)
+
+ExternalProject_Add(fmt_ext
+  GIT_REPOSITORY https://github.com/fmtlib/fmt.git
+  GIT_TAG        9e554999ce02cf86fcdfe74fe740c4fe3f5a56d5
+  INSTALL_DIR    @SMF_DEPS_INSTALL_DIR@
+  CMAKE_ARGS
+    -DCMAKE_BUILD_TYPE=Release
```

Comment:
question. is this not passed in our build. we should pass our `CMAKE_BUILD_TYPE=@SMF_BUILD_TYPE@` thoughts?

Reply:
or we should make a comment that we build all deps in release mode.

Reply:
ahh. yeh, good idea. we'll have to figure out how to handle this for each of the autotools projects

Reply:
I'd say we can do this for the cmake projects to start out and then later we can figure out how to do it for all.

Reply:
That is just to help keep the scope-creep low.

Reply:
fixed. we can use @CMAKE_BUILD_TYPE@ in the CMakeFiles.txt.in template and it will be equal to what we are using in the top-level smf project.